### PR TITLE
feat: Filter activity

### LINKED ISSUE #1595: Filter Activity by type API endpoint
Currently, the API endpoint for user activity retrieves all the activities created in the last 7 days.
This issue is about adding a filter so the list can be retrieved by filtering by one (or more TBD) activity type.

Types:
Proposals Created
Proposals Ended
Votes 
Delegation
Comments
Projects

### PR DESCRIPTION
**Pull request by ncomerci**
**Related Issue**
This closes #1595 

- [ ]  **Functionality validation**
    - [ ]  Feature working as described in the related issue
    - [ ]  New environment variables have been set
    - [ ]  New API services methods have tests
    - [ ]  Code errors are being logged in the code health service
- [ ]  **Product validation**
    - [ ]  UI changes look the same as in Figma
    - [ ]  Usage events are being sent to the product analytics service
    - [ ]  New common components have been documented in Storybook
    - [ ]  Preview link if it’s a customer facing issue
- [ ] **Documentation**
    - [ ] README updated
    - [ ] New API changes are documented
- [ ] **Accessibility**
    - [ ] UI changes are accessible

### CODE REVIEW

File: src/back/routes/events.ts [hunk 1]
```diff
@@ -14,8 +17,17 @@ export default routes((route) => {
...
+async function getLatestEvents(req: Request) {
+  const { event_type } = req.query
+  const filters: Record<string, unknown> = {}
+  if (event_type) {
+    filters.event_type = isArray(event_type) ? event_type : [event_type]
+  }
+  const parsedEventTypes = EventFilterSchema.safeParse(filters)
+  if (!parsedEventTypes.success) {
+    throw new Error('Invalid event types: ' + parsedEventTypes.error.message)
+  }
```
Comment (andyesp) [inline src/back/routes/events.ts:29]: Maybe this could be in a `validateEventTypesFilters` and that function could have a test. Just to be sure.

**Review by 1emu — APPROVED** [review]
LGTM, let's add that tested fn Andy said :)

File: src/back/utils/validations.test.ts [hunk 2]
```diff
@@ -21,3 +23,29 @@ describe('validateProposalId', () => {
...
+    const result = validateEventTypesFilters(req)
+    expect(result).toEqual({ event_type: [EventType.Voted, EventType.Commented] })
+  })
+
+  test('Should throw an error if EventFilterSchema returns an error', () => {
+    const req = { query: { event_type: 'single_event' } } as never
+
+    expect(() => validateEventTypesFilters(req)).toThrow()
+  })
+})
```
Comment (1emu) [inline src/back/utils/validations.test.ts:51]: does the `req` object always has a `query` key? (what if it doesn't? )

File: src/back/utils/validations.test.ts [hunk 3]
```diff
@@ -21,3 +23,29 @@ describe('validateProposalId', () => {
...
+  })
+
+  test('Should keep event_type as an array when it is already an array', () => {
+    const req = { query: { event_type: [EventType.Voted, EventType.Commented] } } as never
+    const result = validateEventTypesFilters(req)
+    expect(result).toEqual({ event_type: [EventType.Voted, EventType.Commented] })
+  })
+
+  test('Should throw an error if EventFilterSchema returns an error', () => {
+    const req = { query: { event_type: 'single_event' } } as never
```
[the remark this answers is not in the document]
Comment (ncomerci) [reply src/back/utils/validations.test.ts:47]: yeap!

File: src/back/utils/validations.test.ts (hunk 2 above)
Comment (ncomerci) [reply src/back/utils/validations.test.ts:51]: By definition in the Request interface, it always has the query object

**Review by 1emu — APPROVED** [review]

**Pull request merged**